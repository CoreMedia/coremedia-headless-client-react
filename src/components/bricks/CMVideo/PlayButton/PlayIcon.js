// @flow
// import PropTypes from 'prop-types';
import styled from 'styled-components';

import { media } from '../../../../styles/themes/utils';

const getEncodedPlayIcon = () => {
  const svg = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
    <circle
      cx="256"
      cy="256"
      r="231"
      fill="rgba(0,0,0,.5)"
      stroke="#fff"
      stroke-width="30"
      stroke-miterlimit="10"
    />
    <path
      d="M348.1 245.6l-134-78.2c-12.2-7.1-22.1-1.4-22 12.7l.7 155.2c.1 14.1 10 19.9 22.3 12.8l133-76.8c12.2-7.1 12.3-18.6 0-25.7z"
      fill="#fff"
    />
  </svg>`;
  return window.btoa(svg);
};

const PlayIcon = styled.div`
  position: absolute;
  bottom: 50%;
  right: 50%;
  transform: translate(50%, 50%);
  z-index: 1;
  width: 90px;
  height: 90px;
  max-width: 75%;
  max-height: 75%;
  background: transparent url(data:image/svg+xml;base64,${getEncodedPlayIcon()}) no-repeat;
  transition: all 0.35s ease(in-out-quad);
  ${props => media(props.theme.breakpoints.desktop)`
		width: 120px;
		height: 120px;
	`};
`;

PlayIcon.displayName = 'PlayIcon';

/*PlayIcon.propTypes = {
  theme: PropTypes.shape({
    breakpoints: PropTypes.shape({
      desktop: PropTypes.number.isRequired,
    }).isRequired,
  }),
};*/

PlayIcon.defaultProps = {
  theme: {
    breakpoints: {
      desktop: 992,
    },
  },
};

export default PlayIcon;
