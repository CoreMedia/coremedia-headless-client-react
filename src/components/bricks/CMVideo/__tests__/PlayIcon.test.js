// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import PlayIcon from '../PlayButton/PlayIcon';

describe('PlayIcon Component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<PlayIcon />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
