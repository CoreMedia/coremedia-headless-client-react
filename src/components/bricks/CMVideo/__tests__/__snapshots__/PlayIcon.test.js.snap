// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayIcon Component should render correctly 1`] = `
.c0 {
  position: absolute;
  bottom: 50%;
  right: 50%;
  -webkit-transform: translate(50%,50%);
  -ms-transform: translate(50%,50%);
  transform: translate(50%,50%);
  z-index: 1;
  width: 90px;
  height: 90px;
  max-width: 75%;
  max-height: 75%;
  background: transparent url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA1MTIgNTEyIj4KICAgIDxjaXJjbGUKICAgICAgY3g9IjI1NiIKICAgICAgY3k9IjI1NiIKICAgICAgcj0iMjMxIgogICAgICBmaWxsPSJyZ2JhKDAsMCwwLC41KSIKICAgICAgc3Ryb2tlPSIjZmZmIgogICAgICBzdHJva2Utd2lkdGg9IjMwIgogICAgICBzdHJva2UtbWl0ZXJsaW1pdD0iMTAiCiAgICAvPgogICAgPHBhdGgKICAgICAgZD0iTTM0OC4xIDI0NS42bC0xMzQtNzguMmMtMTIuMi03LjEtMjIuMS0xLjQtMjIgMTIuN2wuNyAxNTUuMmMuMSAxNC4xIDEwIDE5LjkgMjIuMyAxMi44bDEzMy03Ni44YzEyLjItNy4xIDEyLjMtMTguNiAwLTI1Ljd6IgogICAgICBmaWxsPSIjZmZmIgogICAgLz4KICA8L3N2Zz4=) no-repeat;
  -webkit-transition: all 0.35s ease(in-out-quad);
  transition: all 0.35s ease(in-out-quad);
}

@media (min-width:62em) {
  .c0 {
    width: 120px;
    height: 120px;
  }
}

<div
  className="c0"
/>
`;
